src/models/gerente.model.ts

```diff
@@ -0,0 +1,59 @@
+import { v4 as uuidv4 } from 'uuid';
+import { validaNome } from '../helpers/helpers';
+import { ContaBancaria, TipoConta } from '../models/conta.model';
+import { Cliente } from './cliente.model';
+
+let listaNomesGerentes: string []= [];
+let listaIdsGerentes: string []= [];
+
+export class Gerente {
+    nomeCompleto: string;
+    id: string = '';
+    clientes: Cliente[];
+
+    constructor(nomeCompleto: string, clientes: Cliente[]) {
+        const index = listaNomesGerentes.indexOf(nomeCompleto);
+        if (index !== -1) {
+            this.id = listaIdsGerentes[index];
+        } else {
+            this.id = uuidv4();
+            listaIdsGerentes.push(this.id);
+            listaNomesGerentes.push(nomeCompleto);
+        }
+
+        this.nomeCompleto = nomeCompleto;
+        this.id = uuidv4();
+        this.clientes = clientes;
+
+        const mensagemErroNomeGerente = validaNome(nomeCompleto);
+        if (mensagemErroNomeGerente) {
+            throw new Error(mensagemErroNomeGerente);
+        }
+
+    }
+
+    adicionarCliente(cliente: Cliente): void {
+        this.clientes.push(cliente);
```

Comment:
No arquivo `src/services/cliente.service.ts` você tem uma função de adicionar clientes que faz exatamente a mesma coisa `this.clientes.push(cliente)`. Você pode centralizar a lógica em comum em apenas 1 local e reutilizá-lo!